Automattic/wp-calypso
# Hosting Command Palette: Split the search into label and keywords to display command labels first

Closes https://github.com/Automattic/dotcom-forge/issues/4836

## Proposed Changes

This PR improves the search for the commands:

* if there is a search keyword that matches the label, it will match and display that command to the top:

<img width="1020" alt="Screenshot 2023-12-13 at 3 49 05 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/3e1605b4-aac0-49f8-9f23-eebc5a40d53b">

* if there is a search keyword that matches the `searchLabel`, it will add these commands next:

<img width="1068" alt="Screenshot 2023-12-13 at 3 50 35 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/3a3097e6-7fc9-4ce8-8142-50dbd81a58b0">

* the filtering on the site will remain as is as per the issue description

## Testing Instructions

* Navigate to Calypso Live Link
* Press `cmd + k` on macOS or `ctrl + k` on Windows to trigger command palette 
* Type `reader` in the search field and observe that the command that comes up is `Open reader`
* Type `edge` in the search field and observe that the commands that come up first have `edge` in the command label and the commands that come after are the commands that have `edge` in their `searchLabel`
* Navigate to the nested commands
* Confirm that the sites filtering remains unaffected and works as before

**Notes**
* I am wondering if we should use the `filter` prop that is [already with the `cmdk`](https://www.npmjs.com/package/cmdk#command-cmdk-root) and try to explain the logic there or we should stick with our custom logic - what are your thoughts @sejas ?

## Pre-merge Checklist
